### PR TITLE
docs: Campaigns updates

### PR DESCRIPTION
**Pull request by ryan-blunden**
- Changed content hierarchy so page outline provides a better overview
- Revised section on scalability
- Added section on clearing the cache
- Other content edits

Screen Shot 2020-02-20 at 10 58 48 AM

### CODE REVIEW

**Comment by sqs** [comment]
Rename automation to campaigns 😃 in the PR title, nav bar, etc

**Comment by ryan-blunden** [comment]
> Rename automation to campaigns 😃 in the PR title, nav bar, etc

That's been done, but I haven't changed the name of the doc files as I don't think that's as important.

File: doc/user/automation.md [hunk 1]
```diff
@@ -26,43 +26,49 @@ Without any further configuration Automation is **only be accessible to site-adm
...
 
-Read on for the longer version.
+Read on detailed steps and documentation.
 
-> **Note about scalability**: the patches are generated on the machine on which the `src` CLI tool is being run. That means archives of the repositories in a campaign have to be downloaded from your Sourcegraph instance to the local machine. We're working on remote execution of campaign actions. For now feel free to use a bigger machine, possibly closer to your Sourcegraph instance so that downloading archives of repositories is faster, and use the `-j` parameter to tune the number of parallel jobs being executed.
+## Where to best run campaigns
 
-#### Defining an action
+The patches for a campaign plan are generated on the machine where the `src` CLI is executed, which in turn, downloads zip archives and runs each step against each repository. Therefore, we recommend campaign plans should run (in most cases) on a Linux machine with considerable CPU, RAM, and network bandwidth to reduce the execution time. Putting this machine in the same network as your Sourcegraph instance will also improve performance.
```
Comment (mrnugget) [inline doc/user/automation.md]: At this point in the document the user doesn't know what a "campaign plan" is yet, so I would rephrase:

```suggestion
The patches for a campaign are generated on the machine where the `src` CLI is executed, which in turn, downloads zip archives and runs each step against each repository. For most usecases we recommend that `src` CLI should be run on a Linux machine with considerable CPU, RAM, and network bandwidth to reduce the execution time. Putting this machine in the same network as your Sourcegraph instance will also improve performance.
```

File: doc/user/automation.md [hunk 2]
```diff
@@ -1,14 +1,14 @@
-# Automation
+# Campaigns
 
-> Automation is currently available in private beta for select enterprise customers.
+> Campaigns is currently in private beta for select enterprise customers.
```
Comment (mrnugget) [inline doc/user/automation.md]:
```suggestion
> Campaigns are currently in private beta for select enterprise customers.
```

See [discussion here](https://sourcegraph.slack.com/archives/CMMTWQQ49/p1582186187246300)

File: doc/user/automation.md [hunk 3]
```diff
@@ -26,43 +26,49 @@ Without any further configuration Automation is **only be accessible to site-adm
...
+## Where to best run campaigns
 
-#### Defining an action
+The patches for a campaign plan are generated on the machine where the `src` CLI is executed, which in turn, downloads zip archives and runs each step against each repository. Therefore, we recommend campaign plans should run (in most cases) on a Linux machine with considerable CPU, RAM, and network bandwidth to reduce the execution time. Putting this machine in the same network as your Sourcegraph instance will also improve performance.
 
-The first thing we need is a definition of an "action". An action is what produces a patch and describes what commands or Docker containers to run over which repositories.
+Another factor affecting execution time is the number of jobs executed in parallel, which is by default, the number of cores on the machine. This can be adjusted using the `-j` parameter.
```
Comment (mrnugget) [inline doc/user/automation.md]:
```suggestion
Another factor affecting execution time is the number of jobs executed in parallel, which is by default the number of cores on the machine. This can be adjusted using the `-j` parameter.
```

File: doc/user/automation.md [hunk 4]
```diff
@@ -26,43 +26,49 @@ Without any further configuration Automation is **only be accessible to site-adm
...
 
-> **Note**: At the moment only two `"type"`s of steps are supported: `"docker"` and `"command"`. See `src actions exec -help` for more information.
+To make it simpler for customers, we're [working on remote execution of campaign](https://github.com/sourcegraph/src-cli/pull/128) of campaign actions and would love your feedback.
 
-Here is an example defintion of an action:
+## Defining a campaign action plan
```
Comment (mrnugget) [inline doc/user/automation.md]:
```suggestion
## Defining a action
```

File: doc/user/automation.md [hunk 5]
```diff
@@ -26,43 +26,49 @@ Without any further configuration Automation is **only be accessible to site-adm
...
+
+The first thing we need is a definition of an "action". An action contains a list of steps to run commands or Docker containers to produce a patch for each repository returned by the results of the `scopeQuery` search string.
```
Comment (mrnugget) [inline doc/user/automation.md]:
```suggestion
The first thing we need is a definition of an "action". An action contains a list of steps to run in each repository returned by the results of the `scopeQuery` search string.
```

The next sentence already explains the types of steps.

File: doc/user/automation.md [hunk 6]
```diff
@@ -89,95 +95,116 @@ Here is an example defintion of an action:
 }
 ```
 
-This action runs over every repository that has `go-` in its name and doesn't have an `INSTALL.md` file.
+This action will execute on every repository that has `go-` in its name and doesn't have an `INSTALL.md` file.
```
Comment (mrnugget) [inline doc/user/automation.md]:
```suggestion
Due to the definition of the `"scopeQuery"` this action will execute in every repository that has `go-` in its name and doesn't have an `INSTALL.md` file.
```

File: doc/user/automation.md [hunk 7]
```diff
@@ -89,95 +95,116 @@ Here is an example defintion of an action:
...
+2. The second step, again a `"command"` step, runs the `sed` command to replace text in the `README.md` file in the root of each repository (the `-i ''` argument is only necessary for BSD versions of `sed` that usually come with macOS). Please note that the executed command is simply `sed` which means its arguments are _not_ expanded, as they would be in a shell. To achieve that, execute the `sed` as part of a shell invocation (using `sh -c` and passing in a single argument, for example, like in the first step).
 
-The third step builds a Docker image from the specified `"dockerfile"` and starts a container with this image in which the repository is mounted under `/work`.
+3. The third step builds a Docker image from the specified `"dockerfile"` and starts a container with this image in which the repository is mounted under `/work`.
 
-The fourth step pulls the `golang:1.13-alpine` image from Docker hub, starts a container from it and runs `go fix /work/...` in it.
+4. The fourth step pulls the `golang:1.13-alpine` image from Docker hub, starts a container from it and runs `go fix /work/...` in it.
```
Comment (mrnugget) [inline doc/user/automation.md]: Just realized that we don't do this yet:

```suggestion
4. The fourth step starts a Docker container based on the `golang:1.13-alpine` image and runs `go fix /work/...` in it.
```

File: doc/user/automation.md [hunk 8]
```diff
@@ -89,95 +95,116 @@ Here is an example defintion of an action:
...
+src actions exec -f action.json
 ```
 
 This command is going to:
 
-1. Download or build the required Docker images, if necessary.
-2. Download a ZIP archive of the repositories matched by the `"scopeQuery"` from the Sourcegraph instance to a local temporary directory in `/tmp`.
-3. Execute the action in each repository in parallel (the maximum number of parallel jobs can be configured with `-j`, the default is number of cores on the machine), with each step in an action being executed sequentially on the same copy of a repository. If a step in an action is of type `"command"` the command will be executed in the temporary directory that contains the copy of the repository. If the type is `"docker"` then a container will be launched in which the repository is mounted under `/work`.
-4. Produce a diff for each repository between a fresh copy of the repository's contents and directory in which the action ran.
+1. Pull the required Docker image if necessary.
```
Comment (mrnugget) [inline doc/user/automation.md]: Again: we don't pull images automatically yet

```suggestion
1. Build the required Docker image if necessary.
```

File: doc/user/automation.md [hunk 9]
```diff
@@ -89,95 +95,116 @@ Here is an example defintion of an action:
...
+Edits to the name and description of a campaign can also be made in the web UI with the changes reflected in each changeset. The branch name of a draft campaign with a plan can also be edited, but only if the campaign doesn't contain any published changesets.
 
-Edits to the name and description of a campaign can also be made on the web interface. These changes are then reflected in the changesets by updating them the code hosts. The branch of a draft campaign with a plan can also be edited, but only as long as the campaign doesn't contain any published changesets.
+## Clearing the campaign action cache
 
-## Example: Add a GitHub action to upload LSIF data to Sourcegraph
+Campaign diffs are intelligently cached based on the `scopeQuery` and defined `steps`, but the need to clear the cache to run the steps from scratch may be required.
+
+Clearing the cache requires the manual deletion of the cache directory. If no `-cache` flag is passed to `src actions exec`, the default location of the cache is used which can be found for your platform by running:
```
Comment (mrnugget) [inline doc/user/automation.md]:
```suggestion
Clearing the cache requires either manually emptying the cache directory or using a different one. If no `-cache` flag is passed to `src actions exec`, the default location of the cache is used which can be found for your platform by running:
```

File: doc/user/automation.md [hunk 10]
```diff
@@ -89,95 +95,116 @@ Here is an example defintion of an action:
...
+
+```sh
+src actions exec -help 2>&1 | head -n 3
```
Comment (mrnugget) [inline doc/user/automation.md]:
```suggestion
src actions exec -help
```

The other thing is brittle, since it could break with new flags.

File: doc/user/automation.md [hunk 11]
```diff
@@ -89,95 +95,116 @@ Here is an example defintion of an action:
...
+```
+
+## Example campaigns
+
+The following examples demonstrate various campaign types for different languages using both commands and Docker images. They also provide commentary on considerations such as adjusting the duration (`-timeout`) for actions that exceed the 15 minute default limit.
```
Comment (mrnugget) [inline doc/user/automation.md]:
```suggestion
The following examples demonstrate various types of campaigns for different languages using both commands and Docker images. They also provide commentary on considerations such as adjusting the duration (`-timeout`) for actions that exceed the 15 minute default limit.
```

File: doc/user/automation.md (hunk 9 above)
Comment (ryan-blunden) [reply doc/user/automation.md]: I think this conflates clearing (deleting) the cache and forcing a fresh run by specifying a new and empty cache location but that's a minor concern and isn't worth rewriting at this point. Thanks for the addition.

File: doc/user/automation.md (hunk 4 above)
Comment (ryan-blunden) [reply doc/user/automation.md]:
```suggestion
## Defining an action
```
